packages/variable-editor/src/VariablesEditor.tsx

```diff
@@ -162,3 +164,27 @@ function VariableEditor(props: EditorProps) {
 }
 
 export default VariableEditor;
+
+export const addValidations = (variables: Array<Variable>, validations: ValidationMessages) => {
+  if (validations.length === 0) {
+    return;
+  }
+  const groupedValidations = groupBy(validations, val => val.path);
+  variables.forEach(variable => {
+    const key = variable.name;
+    variable.validations = groupedValidations[key] ?? [];
+    addValidationsRecursive(variable.children, groupedValidations, key);
+  });
```

Comment:
looks douplicated, cant you just call here directly 
```suggestion
  addValidationsRecursive(variables, groupedValidations, '');
```

Reply:
maybe you need to add the dot in the call and not in the function